Plugin/Model/MenuBuilderCommand.php

```diff
@@ -1,20 +1,35 @@
 <?php
+declare(strict_types=1);
+
 namespace RedChamps\CleanMenu\Plugin\Model;
 
 use Magento\Backend\Model\Menu\Builder\AbstractCommand;
-use RedChamps\CleanMenu\Plugin\Base;
+use RedChamps\CleanMenu\Model\Config;
+use function in_array;
 
-class MenuBuilderCommand extends Base
+final class MenuBuilderCommand
 {
-    public function afterExecute(AbstractCommand $subject, $result)
+    /**
+     * @var Config
+     */
+    private $config;
+
+    public function __construct(
+        Config $config
+    ) {
+        $this->config = $config;
+    }
+
+    public function afterExecute(AbstractCommand $subject, $result): array
     {
-        if (isset($result["module"])) {
-            $moduleName = $result["module"];
-            $parentId = isset($result["parent"]) ? $result["parent"] : null;
-            if ($parentId == null && strpos($moduleName, "Magento_") !== 0 && $moduleName != "RedChamps_CleanMenu") {
-                $result["parent"] = self::MENU_ID;
+        if (isset($result['module'])) {
+            $moduleName = $result['module'] ?? '';
+            $parentId = $result['parent'] ?? null;
+            if ($parentId === null && in_array($moduleName, $this->config->getRestrictedModules(), true)) {
```

Comment:
Third-party extension items from Main navigation stopped moving under 'Extensions' menu item & it seems like the problem is on this line. May you please check it.

Reply:
I've set up the configuration to blacklist modules instead of whitelist. Configuration is availabl in the extensions > Settings tab.

Reply:
I believe the setting should be to whitelist the modules. Everyone installing this extension will expect the menu items to be arranged as soon as installed. The manual configuration should be to adjust its behavior. 

I hope this makes sense.

Reply:
Indeed as it's the orignal behavior of the module, it does make sense.
Also I've added the Admin menu cache flush on the save config.

Reply:
Awesome! I am going to test the changes tomorrow and will process the pull request if everything works well.

Reply:
**Regarding "Admin menu cache flush on the save config":** This isn't working when you change the setting and later check "use system value" & save the config again. i.e it works correctly when you change the setting normally but not working when you check "use system value" & save the config